### PR TITLE
Send SYN retries to the same backend.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Allow conntrack lookup for SYNs.

Fixes https://github.com/projectcalico/calico/issues/4290

## Todos
- [x] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
In eBPF mode, ensure that SYN retries to a NATted destination go to the same backing workload to prevent spurious RSTs after a SYN-ACK is lost.
```

### CODE REVIEW

**Review by tomastigera — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
